### ISSUE TITLE
AlignTemplatedTestCases does not handle [Template] NONE as expected

### ISSUE THREAD

**Issue by admorgan** [MarketSquare/robotframework-tidy#444]
If a test has set the template to none, I would not expect it to be formatted the same as the templates with a value. Even if the template has a value I would expect that the [] sections would not be aligned to the columns taking data. This is larger part of what is described in #443.

Example input:
```
Test Template       Test Commit Message

*** Test Cases ***    UNLINTED FILE    LINTED FILE
Misplaced Git Trailer    misplaced_trailer.txt    misplaced_trailer_linted.txt
Mixed Unordered List Symbols    mixed_unordered.txt    mixed_unordered_linted.txt
Incorrectly Ordered List    ordered.txt    ordered_linted.txt
Unaligned List Item Indention    unaligned.txt    unaligned_linted.txt
Garbage Commit Message    garbage.txt    garbage_linted.txt
Valid Commit Message    unchanged.txt    unchanged_linted.txt    ${type trailer}    ${type trailer}\n${trailer block}
Header Validation
    [Template]    NONE
    ${valid header} =    Validate Header    ${header}
    Should Be True    ${valid header}
    ${err invalid header} =    Validate Header    ${EMPTY}
    Should Not Be True    ${err invalid header}
Trailer Validation
    [Template]    NONE
    ${valid type trailer} =    Validate Type Trailer    ${type trailer}
    Should Be True    ${valid type trailer}
    ${err invalid trailer} =    Validate Type Trailer    ${EMPTY}
    Should Not Be True    ${err invalid trailer}
```

Generated output:
```
Test Template       Test Commit Message

*** Test Cases ***                  UNLINTED FILE               LINTED FILE
Misplaced Git Trailer               misplaced_trailer.txt       misplaced_trailer_linted.txt
Mixed Unordered List Symbols        mixed_unordered.txt         mixed_unordered_linted.txt
Incorrectly Ordered List            ordered.txt                 ordered_linted.txt
Unaligned List Item Indention       unaligned.txt               unaligned_linted.txt
Garbage Commit Message              garbage.txt                 garbage_linted.txt
Valid Commit Message                unchanged.txt               unchanged_linted.txt            ${type trailer}     ${type trailer}\n${trailer block}
Header Validation
                                    [Template]                  NONE
                                    ${valid header}=            Validate Header                 ${header}
                                    Should Be True              ${valid header}
                                    ${err invalid header}=      Validate Header                 ${EMPTY}
                                    Should Not Be True          ${err invalid header}
Trailer Validation
                                    [Template]                  NONE
                                    ${valid type trailer}=      Validate Type Trailer           ${type trailer}
                                    Should Be True              ${valid type trailer}
                                    ${err invalid trailer}=     Validate Type Trailer           ${EMPTY}
                                    Should Not Be True          ${err invalid trailer}
```

Expected output:
```
Test Template       Test Commit Message

*** Test Cases ***                  UNLINTED FILE               LINTED FILE
Misplaced Git Trailer               misplaced_trailer.txt       misplaced_trailer_linted.txt
Mixed Unordered List Symbols        mixed_unordered.txt         mixed_unordered_linted.txt
Incorrectly Ordered List            ordered.txt                 ordered_linted.txt
Unaligned List Item Indention       unaligned.txt               unaligned_linted.txt
Garbage Commit Message              garbage.txt                 garbage_linted.txt
Valid Commit Message                unchanged.txt               unchanged_linted.txt            ${type trailer}     ${type trailer}\n${trailer block}
Header Validation
    [Template]    NONE
    ${valid header} =    Validate Header    ${header}
    Should Be True    ${valid header}
    ${err invalid header} =    Validate Header    ${EMPTY}
    Should Not Be True    ${err invalid header}
Trailer Validation
    [Template]    NONE
    ${valid type trailer} =    Validate Type Trailer    ${type trailer}
    Should Be True    ${valid type trailer}
    ${err invalid trailer} =    Validate Type Trailer    ${EMPTY}
    Should Not Be True    ${err invalid trailer}                              
```

**Comment by bhirsz** [comment]
Interesting case and it totally makes sense. For sure we can just disable aligner if the template is set to be none. There is also option to allow such tests to be aligned by aligner for non templated tests. I will see what can be done.

**Comment by bhirsz** [comment]
Fix is included in just released Robotidy 4.0! However I recommend waiting till the Robotidy 4.0.1 is (soon) released because I introduced regression.
